dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb

```
@@ -196,7 +196,11 @@ jenkins:
       <%- k8s_setup['agent_definitions'].each do |agent| -%>
         - containers:
           - alwaysPullImage: false
+          <%- if agent['imageName'] && !agent['imageName'].empty? -%>
+            image: "<%= agent['imageName'] %>"
+          <%- else -%>
             image: "<%= @jcasc['agent_images']['container_images'][agent['name']] %>"
+          <%- end -%>
```

```suggestion
            image: "<%= agent['imageName'] && !agent['imageName'].empty? ? agent['imageName'] : @jcasc['agent_images']['container_images'][agent['name']] %>"
```

Personal preference: in this context (a template generating a scalar YAML key/value) I prefer the ternary notation (to avoid risk of messing up the whitespace and tabs with the `<%- -%>` brackets